pkg/deploy/controller/deployerpod/controller.go

```diff
@@ -123,68 +114,6 @@ func (c *DeployerPodController) Handle(pod *kapi.Pod) error {
 	return nil
 }
 
-func (c *DeployerPodController) cleanupFailedDeployment(deployment *kapi.ReplicationController) error {
```

Comment:
This is probably obvious if you know the controller better, but let's say I have a DC that has a PodSpec that grabs a NodePort.  The post-deploy hook fails.  Does this leave me with an running Pod that is claiming that NodePort instead of cleaning it up?